src/arb_mat/vector_mul.c

```diff
@@ -0,0 +1,62 @@
+/*
+    Copyright (C) 2023 Jean Kieffer
+
+    This file is part of Arb.
+
+    Arb is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "arb_mat.h"
+
+void
+arb_mat_vector_mul_row(arb_ptr res, arb_srcptr v, const arb_mat_t A, slong prec)
```

Comment:
These functions could be implemented more efficiently: by making shallow copies and calling ``arb_dot`` directly, one should be able to get away with a single, shallow temporary vector and 0 calls to ``arb_set``. There could be some aliasing pitfalls though. It's up to you if you want to spend time on this.

Reply:
In https://github.com/flintlib/flint/pull/1524/commits/e4cafd2fa5f95468b60811afc5d8050b57b21ef2 I wrote underscore versions that don't allow aliasing using `arb_dot` only, and a non-underscore version that also does `arb_set`'s.
Do you think it would it be better to follow the naming conventions (or way of testing) of `fmpz_mat_mul_fmpz_vec` here? It doesn't look like that function allows any aliasing.

Reply:
The row version acessing A->entries isn't safe, because the rows may be in a scrambled order, and for window matrices A->entries doesn't even exist. You'll need a shallow transpose, which can be done column by column. See how acb_mat_solve_tril_classical does it.

I have no strong opinion about the naming!

Reply:
This should be fixed in https://github.com/flintlib/flint/pull/1524/commits/a567243db9119183bb50d2aa98692041caef16ec.